lib/storage/metadata/proxy/BucketdRoutes.js

```diff
@@ -32,15 +32,19 @@ class BucketdRoutes {
 
     _getBucketAttributes(req, res, bucketName, logger) {
         return this._metadataWrapper.client.getBucketAttributes(
-            bucketName, logger, (err, data) =>
-                                sendResponse(req, res, logger, err,
-                                    BucketInfo.fromObj(data).serialize())
-                            );
+            bucketName, logger, (err, data) => {
+                if (data === undefined) {
+                    return sendResponse(req, res, logger,
+                                        errors.NoSuchBucket);
+                }
+                return sendResponse(req, res, logger, err,
```

Comment:
If there was a TCP error making a request, would data be `undefined` or `null` and `err` be set?

Reply:
err and data come from the MetadataWrapper and are not supposed to carry information about the underlying TCP connection, if any.
But I might want to check for an error before checking the payload validity

Reply:
Since we follow error first callbacks style, it's inline.